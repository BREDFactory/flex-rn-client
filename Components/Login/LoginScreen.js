// @flow

import React from 'react'
import { AsyncStorage, View } from 'react-native'

import { Text } from 'react-native-elements'
import { omit } from 'ramda'
import styles from './LoginScreenStyles'
import server from '../../config/server'
import config from '../../config/api'
import type { Props, State } from './LoginScreenType'

import { config_regex } from '../../config/regex.json'

import I18n from '../../i18n/i18n'
import LoginButton from './components/LoginButton'
import InputLogin from './components/InputLogin'

class LoginScreen extends React.Component<Props, State> {
  static navigationOptions = {
    title: I18n.t('login.title')
  }

  constructor() {
    super()
    this.state = {
      name: '',
      fname: '',
      id: '',
      place: '',
      debug: '',
      debugField: '',
      historical: []
    }
  }

  logOut() {
    AsyncStorage.removeItem('USER')
  }

  /** This function handle the user login */
  logIn() {
    const { navigation } = this.props

    if (
      this.state.name !== '' &&
      this.state.fname !== '' &&
      this.state.id !== '' &&
      this.state.id.match(config_regex) !== null
    ) {
      const payload = {
        name: this.state.name,
        fname: this.state.fname,
        id_user: this.state.id,
        id_place: '',
        historical: this.state.historical
      }

      
      fetch(`${server.address}login_user`, {
        method: 'POST',
        body: JSON.stringify(payload),
        headers: {
          'Content-Type': 'application/json',
          'x-access-token': config.token
        }
      })
        .catch(err => console.log(err))
        .then(res => res.json())
        .then(data => {
          const redirect: boolean = true
          if (redirect) {
            if (data.user) this.setState({ isRemote: data.user.isRemote })
            AsyncStorage.setItem(
              'USER',
              JSON.stringify(omit('debugField', this.state))
            )
            navigation.goBack()
            navigation.navigate('Profile')
          }
        })
    } else {
      this.setState({ debugField: I18n.t('login.debug') })
    }
  }

  render() {
    const { debugField } = this.state
    return (
      <View style={styles.view}>
        <View style={styles.view_second}>
          <InputLogin onChangeText={text => this.setState({ name: text })}
                      onChangeText1={text => this.setState({ fname: text })}
                      onChangeText2={text => this.setState({ id: text })}/>
          <LoginButton onPress={() => this.logIn()}/>

          <Text style={styles.debug}>{debugField}</Text>
        </View>
      </View>
    )
  }
}

export default LoginScreen
