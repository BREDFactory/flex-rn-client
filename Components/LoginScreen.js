import React from 'react';
import { View, TextInput, Button, Text } from 'react-native';
import { AsyncStorage } from 'react-native';

class LoginScreen extends React.Component {
  static navigationOptions = {
    title: 'Login',
  };

  constructor(){
    super();
    this.state = {
      name: '',
      fname: '',
      id:'',
      place:'',
      debug:'',
    }
  }

  _logOut() {
    AsyncStorage.removeItem('USER');
  }

  _logIn() {
    if(this.state.name != '' && this.state.fname != '' && this.state.id != '')
    {
      AsyncStorage.setItem('USER', JSON.stringify(this.state));
      this.setState({debug:'OK'});
      const navigation = this.props.navigation;
      navigation.goBack();
      navigation.navigate('Profile');
    }
    else {
      this.setState({debug:'Fill all inputs'});
    }
  }

  render() {
    return (
      <View style={{
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
      }}>
        <View style={{
          width:250,
          marginBottom:50,
        }}>
          <TextInput style={{
            borderRadius: 4,
            borderWidth: 0.5,
            margin:5,
          }}
           placeholder="Nom"
           onChangeText={(text) => this.setState({name:text})}
          />

          <TextInput style={{
            borderRadius: 4,
            borderWidth: 0.5,
            margin:5,
          }}
            placeholder="PrÃ©nom"
            onChangeText={(text) => this.setState({fname:text})}
          />

          <TextInput style={{
            borderRadius: 4,
            borderWidth: 0.5,
            margin:5,
          }}
            placeholder="ID"
            onChangeText={(text) => this.setState({id:text})}
          />
          <View style={{
            marginTop:10,
          }}>
            <Button title='Login'
            color="#000"
            onPress={() => this._logIn()}/>
          </View>


          <Text>{this.state.debug}</Text>
        </View>
      </View>
    );
  }
}
/*<Button title='Log out'
onPress={() => this._logOut()}/>*/
export default LoginScreen
