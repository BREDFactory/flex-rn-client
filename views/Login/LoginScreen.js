/*
Copyright 2019-2020 BRED Banque Populaire

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
// @flow

import React from "react";
import { AsyncStorage, View, Animated, Keyboard } from "react-native";

import { Text } from "react-native-elements";
import { omit } from "ramda";
import LinearGradient from "react-native-linear-gradient";
import styles from "./LoginScreenStyles";
import server from "../../config/server";
import config from "../../config/api";
import regex from "../../config/regex";
import type { Props, State } from "./LoginScreenType";

import logo from "../../assets/logo.png";

import I18n from "../../i18n/i18n";
import { checkNavigation } from "../../utils/utils";

/**
 * List of components
 */
// eslint-disable-next-line
import LoginButton from "@components/Login/LoginButton";
// eslint-disable-next-line
import InputLogin from "@components/Login/InputLogin";

class LoginScreen extends React.Component<Props, State> {
  static navigationOptions = {
    header: (
      <View
        style={{
          paddingTop: 20 /* only for IOS to give StatusBar Space */,
          backgroundColor: "white",
          height: 80
        }}
      >
        <View
          style={{
            flex: 1,
            alignItems: "center",
            justifyContent: "center"
          }}
        >
          <Text
            style={{
              color: "black",
              fontWeight: "bold",
              fontSize: 20,
              textAlign: "center",
              fontFamily: "Raleway"
            }}
          >
            Flex-Office
          </Text>
        </View>
        <LinearGradient
          start={{ x: 0, y: 0 }}
          end={{ x: 1, y: 0 }}
          colors={["#58C0D0", "#468BB6", "#3662A0"]}
          style={{ width: "100%", height: 7 }}
        />
      </View>
    )
  };

  constructor() {
    super();
    this.state = {
      name: "",
      fname: "",
      id: "",
      place: "",
      debug: "",
      debugField: "",
      remoteDay: "",
      historical: [],
      photo: ""
    };
    // Animation values for iOs (only) keyboard handling
    this.keyboardHeight = new Animated.Value(0);
    this.imageHeight = new Animated.Value(120);
  }

  componentWillMount() {
    checkNavigation(this);
    
    // Keyboard listeners for iOs only
    this.keyboardWillShowSub = Keyboard.addListener('keyboardWillShow', this.keyboardWillShow);
    this.keyboardWillHideSub = Keyboard.addListener('keyboardWillHide', this.keyboardWillHide);
  }

  componentWillUnmount() {
    this.keyboardWillShowSub.remove();
    this.keyboardWillHideSub.remove();
  }

  // Keyboard event handlers for iOs only
  keyboardWillShow = (event) => {
    Animated.parallel([
      Animated.timing(this.keyboardHeight, {
        duration: event.duration,
        toValue: event.endCoordinates.height,
      }),
      Animated.timing(this.imageHeight, {
        duration: event.duration,
        toValue: 40
      }),
    ]).start();
  };

  keyboardWillHide = (event) => {
    Animated.parallel([
      Animated.timing(this.keyboardHeight, {
        duration: event.duration,
        toValue: 0,
      }),
      Animated.timing(this.imageHeight, {
        duration: event.duration,
        toValue: 120,
      }),
    ]).start();
  };

  /** This function handle the user login */
  logIn() {
    const { navigation } = this.props;
    const { name, fname, id, historical } = this.state;

    if (
      name !== "" &&
      fname !== "" &&
      id !== "" &&
      id.match(regex.login_regex) !== null
    ) {
      const payload = {
        name,
        fname,
        id_user: id,
        id_place: "",
        historical
      };

      fetch(`${server.address}login_user`, {
        method: "POST",
        body: JSON.stringify(payload),
        headers: {
          "Content-Type": "application/json",
          "x-access-token": config.token
        }
      })
        .then(res => {
          if (res.status === 200) return res.json();
          return false;
        })
        .then(data => {
          if (data !== false) {
            if (data.user)
              this.setState({
                remoteDay: data.user.remoteDay,
                photo: data.user.photo,
                friend: data.user.friend,
                place: data.user.id_place
              });
            else this.setState({ friend: [] });
            AsyncStorage.setItem(
              "USER",
              JSON.stringify(omit(["debugField"], this.state))
            );
            navigation.navigate(
              "Profile",
              data.user ? { photo: data.user.photo } : {}
            );
          } else {
            this.setState({ debugField: I18n.t("login.debug") });
          }
        });
    } else {
      this.setState({ debugField: I18n.t("login.debug") });
    }
  }

  render() {
    const { debugField } = this.state;
    return (
      // Animation will work only on iOs
      <Animated.View style={ [styles.view, { paddingBottom: this.keyboardHeight }] }>
        <Animated.Image source={logo} style={{ height: this.imageHeight, resizeMode: "contain" }} />
        <View style={styles.view_second}>
          <InputLogin
            onSubmitEditing={() => this.logIn()}
            onChangeText={text => this.setState({ name: text })}
            onChangeText1={text => this.setState({ fname: text })}
            onChangeText2={text => this.setState({ id: text })}
          />
          <LoginButton onPress={() => this.logIn()} />
          <Text style={styles.debug}>{debugField}</Text>
        </View>
      </Animated.View>
    );
  }
}

export default LoginScreen;
