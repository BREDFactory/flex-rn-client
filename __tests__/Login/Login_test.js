import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import LoginScreen from "../../Components/Login/LoginScreen";
import "isomorphic-fetch";

const navigation = { navigate: jest.fn(), popToTop: jest.fn() };

it("renders correctly", () => {
  const tree = renderer
    .create(<LoginScreen navigation={navigation} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
