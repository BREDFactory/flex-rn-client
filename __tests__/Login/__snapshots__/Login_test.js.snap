// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 50,
        "width": 250,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "borderBottomColor": "#bdc6cf",
            "borderBottomWidth": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Nom"
          style={
            Array [
              Object {
                "color": "#86939e",
                "minHeight": 36,
                "width": 750,
              },
              Object {
                "fontSize": 17.5,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#bdc6cf",
            "borderBottomWidth": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="PrÃ©nom"
          style={
            Array [
              Object {
                "color": "#86939e",
                "minHeight": 36,
                "width": 750,
              },
              Object {
                "fontSize": 17.5,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#bdc6cf",
            "borderBottomWidth": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="ID"
          style={
            Array [
              Object {
                "color": "#86939e",
                "minHeight": 36,
                "width": 750,
              },
              Object {
                "fontSize": 17.5,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            undefined,
            Object {
              "borderRadius": 15,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#5167A4",
                },
                Object {
                  "borderRadius": 15,
                },
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#fff",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              S'identifier
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "color": "#5167A4",
            "margin": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
